alpha/radio.cpp

```diff
@@ -0,0 +1,77 @@
+#include "radio.h"
+#include "esp_pm.h"
+
+// Lower level library include decisions go here
+
+#ifdef HAL_M5STICK_C
+#include <M5StickC.h>
+#include "AXP192.h"
+
+#include <WiFi.h>
+#define WIFI_SSID "Test"       // Enter your SSID here
+#define WIFI_PASS "password"    // Enter your WiFi password here
+#define SSID_DISPLAY_COUNT 3
+#elif HAL_M5STACK
+#include <M5Stack.h>
+#endif
+
+#define ENTER_CRITICAL  xSemaphoreTake(halMutex, portMAX_DELAY)
+#define EXIT_CRITICAL   xSemaphoreGive(halMutex)
+static SemaphoreHandle_t halMutex;
+
+// Your one and only
+_TS_RADIO TS_RADIO;
+_TS_RADIO::_TS_RADIO() {}
+
+void _TS_RADIO::begin()
+{
+  this->wifiInitialized = false;
+}
+
+//
+// WIFI setup
+// 
+void _TS_RADIO::wifi_connect()
+{
+  if (!this->wifiInitialized)
+  {
+    WiFi.mode(WIFI_STA);
+    WiFi.disconnect();
+    int SSID_COUNT = 0;
+    SSID_COUNT = WiFi.scanNetworks();
```

Comment:
I don't understand, whats the purpose of this?

Should it look for WIFI_SSID and only connect if it exists/visible?

How long does scanNetworks take?

Reply:
wifi_connect primarily is meant to find available networks, then connect to a preset one, can be simplifed.
It includes printing to console to display which SSIDs are available, when not available, to print an error.

scanNetworks is almost instant, but I found that having at least a one second wait gives it some buffer to be more reliable